modules/base/src/io/ivfvolumereader.cpp

```diff
@@ -56,97 +59,253 @@
+            TxElement byteOrder{"ByteOrder"};
+
+            if (auto content = littleEndianNode->Attribute("content")) {
+                byteOrder.SetAttribute("content", content == "1" ? "LittleEndian" : "BigEndian");
+            } else {
+                byteOrder.SetAttribute("content", "LittleEndian");
+            }
+            root->InsertBeforeChild(metaDataMap, byteOrder);
+            metaDataMap->RemoveChild(littleEndianNode->Parent());
+        }
+    }
 }
 
-IvfVolumeReader* IvfVolumeReader::clone() const { return new IvfVolumeReader(*this); }
+bool Converter::convert(TxElement* root) {
+    if (root->Value() != InviwoVolume) {
+        // ignore version number if the root node is not already an InviwoVolume, but for example an
+        // InviwoWorkspace
+        version_ = 1;
+    }
 
-std::shared_ptr<Volume> IvfVolumeReader::readData(const std::filesystem::path& filePath) {
-    const auto localPath = downloadAndCacheIfUrl(filePath);
+    switch (version_) {
+        case 0:
+        case 1: {
+            auto metaDataMap = xml::getElement(root, "MetaDataMap");
 
-    checkExists(localPath);
+            if (metaDataMap) {
+                if (auto littleEndianNode =
+                        xml::getElement(metaDataMap, "MetaDataItem&key=LittleEndian/MetaData")) {
+                    // move endianness to own tag
+                    TxElement byteOrder{"ByteOrder"};
+
+                    auto content = littleEndianNode->Attribute("content").value_or("1");
+                    byteOrder.SetAttribute(
+                        "content",
+                        fmt::format("{}", content == "1" ? static_cast<int>(ByteOrder::LittleEndian)
+                                                         : static_cast<int>(ByteOrder::BigEndian)));
+
+                    root->InsertBeforeChild(metaDataMap, byteOrder);
+                    metaDataMap->RemoveChild(littleEndianNode->Parent());
+                }
+            }
+            if (auto rawFile = xml::getElement(root, "RawFile"); rawFile && metaDataMap) {
+                // move metadata map to RawFile
+                rawFile->InsertEndChild(*metaDataMap);
+                root->RemoveChild(metaDataMap);
+            }
+
+            return true;
+        }
+
+        default:
+            return false;  // No changes
+    }
+}
+
+DataMapper deserializeDataMap(Deserializer& d) {
+    DataMapper dataMap;
+    d.deserialize("DataRange", dataMap.dataRange);
+    d.deserialize("ValueRange", dataMap.valueRange);
+    d.deserialize("ValueName", dataMap.valueAxis.name);
+
+    std::string tmp;
+    d.deserialize("ValueUnit", tmp);
+    if (!tmp.empty()) {
+        dataMap.valueAxis.unit = units::unit_from_string(tmp);
+    }
+
+    return dataMap;
+}
+
+std::array<Axis, 3> deserializeAxes(Deserializer& d) {
+    std::array<Axis, 3> axes;
+    std::string tmp;
+    d.deserialize("Axis1Unit", tmp);
+    if (!tmp.empty()) {
+        axes[0].unit = units::unit_from_string(tmp);
+    }
```

Comment:
this begs for a function... 
does unit_from_string not work with an empty string?